### PR TITLE
Added extra dependencies' for python package py-anuga

### PR DESCRIPTION
**Pull request by samcom12**
Added extra dependencies for `py-anuga` and also support for Python3

### CODE REVIEW

**Comment by adamjstewart** [comment]
Flake8:
```
var/spack/repos/builtin/packages/py-anuga/package.py:5: [E265] block comment should start with '# '
var/spack/repos/builtin/packages/py-anuga/package.py:9: [E302] expected 2 blank lines, found 1
var/spack/repos/builtin/packages/py-pymetis/package.py:8: [E303] too many blank lines (3)
var/spack/repos/builtin/packages/py-pymetis/package.py:33: [W291] trailing whitespace
var/spack/repos/builtin/packages/py-triangle/package.py:5: [E265] block comment should start with '# '
var/spack/repos/builtin/packages/py-triangle/package.py:40: [W293] blank line contains whitespace
var/spack/repos/builtin/packages/py-triangle/package.py:41: [W191] indentation contains tabs
var/spack/repos/builtin/packages/py-triangle/package.py:41: [E101] indentation contains mixed spaces and tabs
var/spack/repos/builtin/packages/py-triangle/package.py:41: [E999] TabError: inconsistent use of tabs and spaces in indentation
var/spack/repos/builtin/packages/py-triangle/package.py:41: [E111] indentation is not a multiple of 4
var/spack/repos/builtin/packages/py-triangle/package.py:41: [E113] unexpected indentation
```

File: var/spack/repos/builtin/packages/py-pymetis/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-pymetis/package.py:1]: Please change `2020` to `2021`.

**Comment by adamjstewart** [comment]
Ping @samcom12, are you still working on this?

**Comment by samcom12** [comment]
Will open separate PR for each package.

**Pull request closed without merging**